web-ui/components/Editor.vue

```diff
@@ -57,10 +70,49 @@ export default {
       },
     };
   },
+  mounted: function () {
+    let windowWidth = window.innerWidth || document.documentElement.clientWidth || document.body.clientWidth;
+    this.percentInPx = windowWidth / 100;
+    window.addEventListener("mousemove", (e) => this.resizeEditor(e));
+    window.addEventListener("click", this.stopDrag);
+    window.addEventListener('resize', () => this.updateWindowDimensions());
+  },
+  beforeDestroy() {
+    window.removeEventListener("mousemove", (e) => this.resizeEditor(e));
+    window.removeEventListener("click", this.stopDrag);
+    window.removeEventListener('resize', () => this.updateWindowDimensions());
+  },
   methods: {
     handleTab(tab) {
-      this.tab = tab;
+      this.tab = tab; 
     },
+    updateWindowDimensions(e) {
+      let windowWidth = window.innerWidth || document.documentElement.clientWidth || document.body.clientWidth;
+      this.percentInPx = windowWidth / 100;
+    },
+    startDrag() {
+      this.isResizing = true;
+    },
+    stopDrag() {
+      if(this.isResizing)
+        document.documentElement.style.setProperty('--output-width', `${100 - this.mouseCurrentX/this.percentInPx}%`);
+      this.isResizing = false;
+    },
+    resizeEditor(e) {
+      this.mouseCurrentX = e.clientX;
+      let x = e.clientX;
+      let diff = 0;
+      if(!this.editorResizePx) {
+        this.editorResizePx = x;
+      } else {
+        diff = x - this.editorResizePx;
+        this.editorResizePx = x;
+      }
+      let currentEditorWidth = getComputedStyle(document.documentElement,null).getPropertyValue('--output-width');
+
+      if(this.isResizing && currentEditorWidth.replace('%', '') - diff/this.percentInPx > 0) 
+        document.documentElement.style.setProperty('--output-width', `${currentEditorWidth.replace('%', '') - diff/this.percentInPx}%`);
+    }
```

Comment:
The resize calculation does not need to be this complex. You seem to be assuming that we need to keep the value in a %. We do not, we can convert it to any unit we want, the % just gives us a nice default value.

As such all we really need is to get the current window width, take it away from the current cursor position, bounds limit it and set the width to that value:
```js
      let width = window.innerWidth - e.clientX;

      if (width > window.innerWidth) {
        width = window.innerWidth;
      } else if (width < 0) {
        width = 0;
      }

      document.documentElement.style.setProperty(
        "--output-width",
        `${width}px`,
      );
```